File: src/main/java/com/github/hongkaiwen/netty/play/http/HttpServer.java
package com.github.hongkaiwen.netty.play.http;

import io.netty.bootstrap.ServerBootstrap;
import io.netty.channel.ChannelFuture;
import io.netty.channel.ChannelOption;
import io.netty.channel.nio.NioEventLoopGroup;
import io.netty.channel.socket.nio.NioServerSocketChannel;

/**
 * https://www.cnblogs.com/demingblog/p/9970772.html
 */
public class HttpServer {

  public static void main(String[] args) throws InterruptedException {

    NioEventLoopGroup boss = new NioEventLoopGroup();
    NioEventLoopGroup worker = new NioEventLoopGroup();
    try{
      ServerBootstrap server = new ServerBootstrap();
      server.group(boss, worker)
        .channel(NioServerSocketChannel.class)
        .childHandler(new HttpServerInitiallizer())
        .option(ChannelOption.SO_BACKLOG, 128)
        .childOption(ChannelOption.SO_KEEPALIVE, true);
      ChannelFuture f = server.bind(8080).sync();
      f.channel().closeFuture().sync();
    } finally {
      worker.shutdownGracefully();
      boss.shutdownGracefully();
    }
  }
}

File: src/main/kotllin/com/github/hongkaiwen/netty/play/kt/http/HttpServer.kt
package com.github.hongkaiwen.netty.play.kt.http

import io.netty.bootstrap.ServerBootstrap
import io.netty.channel.ChannelOption
import io.netty.channel.nio.NioEventLoopGroup
import io.netty.channel.socket.nio.NioServerSocketChannel

fun main(args: Array<String>) {
    var boss = NioEventLoopGroup()
    var worker = NioEventLoopGroup()
    try {
        var server = ServerBootstrap()
        server.group(boss, worker)
        server.channel(NioServerSocketChannel::class.java)
        server.childHandler(HttpServerInitiallizer())
        server.option(ChannelOption.SO_BACKLOG, 128)
        server.childOption(ChannelOption.SO_KEEPALIVE, true)
        var cf = server.bind(8888).sync()
        cf.channel().closeFuture().sync()
    }finally {
        boss.shutdownGracefully()
        worker.shutdownGracefully()
    }
}